websites/X/Xoul - AI/presence.ts

```diff
@@ -0,0 +1,112 @@
+const presence = new Presence({
+		clientId: "1310165299802411078",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets {
+	Logo = "https://i.imgur.com/aTWK1eZ.png",
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: Assets.Logo,
+			startTimestamp: browsingTimestamp,
+		},
+		[time, buttons] = await Promise.all([
+			presence.getSetting<boolean>("time"),
+			presence.getSetting<boolean>("buttons"),
+		]),
+		{ pathname, href } = document.location,
+		button = {
+			label: "View Page",
+			url: href,
+		};
+
+	switch (pathname.split("/")[1]) {
+		case "": {
+			presenceData.details = "Browsing Homepage";
+			presenceData.buttons = [button];
+			break;
+		}
+		case "chats": {
+			const chatName = document.querySelector("[class^='ChatUI_xoul_name']");
+			if (chatName)
+				presenceData.details = `Chatting with ${chatName.textContent.trim()}`;
+
+			const chatImage = document.querySelector(
+				"[class^='ChatBubble_image']"
+			) as HTMLImageElement;
+			if (chatImage) presenceData.largeImageKey = chatImage?.src;
+			else presenceData.details = "Chatting";
+
+			break;
+		}
+		case "profile": {
+			const { pathname } = document.location;
```

Comment:
remove

Reply:
I just removed anything relating to what profile they're looking at to avoid privacy concerns, it just says "Viewing a profile" now.